### PR TITLE
Show map name when reporting map parsing errors

### PR DESCRIPTION
**Pull request by DanVanAtta**
…s are passed, we will know which erros correspond to which maps.

Note, this is a bit brute force of a fix. To refactor things for this to be a clean fix would be quite the undertaking, and would lead down a number of different refactor paths that would touch a lot of code. So for this PR, doing the necessary to wire through the additional map name data. 

Here is an example game parse error message after this update (note, before there was no map name in this error message, so it was mysterious which map had the problem):

```
Exception: Steampunk 1915:The following option name of techAbilityAttachment of class TechAbilityAttachment are either misspelled or exist only in a future version of TripleA. Setter: productionBonus
games.strategy.engine.data.GameParseException: Steampunk 1915:The following option name of techAbilityAttachment of class TechAbilityAttachment are either misspelled or exist only in a future version of TripleA. Setter: productionBonus
	at games.strategy.engine.data.GameParser.setValues(GameParser.java:1365)
	at games.strategy.engine.data.GameParser.parseAttachments(GameParser.java:1310)
	at games.strategy.engine.data.GameParser.parse(GameParser.java:154)
	at games.strategy.engine.framework.ui.NewGameChooserEntry.<init>(NewGameChooserEntry.java:55)
```

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/970)

### CODE REVIEW

File: src/games/strategy/engine/data/GameParseException.java [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
   private static final long serialVersionUID = 4015574053053781872L;
 
-  public GameParseException(final String error) {
-    super(error);
+  public GameParseException(final GameData gameData, final String error) {
+    this(gameData.getGameName(), error);
+  }
+
+  public GameParseException(final String mapName, final String error) {
+    super(Joiner.on(':').join(mapName, error).toString());
```
Comment (ron-murhammer) [inline src/games/strategy/engine/data/GameParseException.java]: Is toString() needed here?

File: src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java [hunk 2]
```diff
@@ -326,7 +326,7 @@ private NewGameChooserEntry selectByName(final Component ui, final boolean force
     final String userPreferredDefaultGameName =
         (forceFactoryDefault ? DEFAULT_GAME_NAME : prefs.get(DEFAULT_GAME_NAME_PREF, DEFAULT_GAME_NAME));
     final NewGameChooserModel model = NewGameChooser.getNewGameChooserModel();
-    selectedGame = model.findByName(userPreferredDefaultGameName);
+//    selectedGame = model.findByName(userPreferredDefaultGameName);
```
Comment (ron-murhammer) [inline src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: This is a separate change fixing selecting the default map?

File: src/games/strategy/triplea/attachments/TechAbilityAttachment.java [hunk 3]
```diff
@@ -101,8 +101,11 @@ public static TechAbilityAttachment get(final TechAdvance type, final String nam
   private IntegerMap<UnitType> m_defenseRollsBonus = new IntegerMap<>();
   private IntegerMap<UnitType> m_bombingBonus = new IntegerMap<>();
 
+  private final String mapName;
```
Comment (ron-murhammer) [inline src/games/strategy/triplea/attachments/TechAbilityAttachment.java]: Is this field needed? Looks like some of the exception calls pass data and others pass mapName?

File: src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java (hunk 2 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/startup/mc/GameSelectorModel.java]: No, was testing out the default map selection logic. Which now assumes a map that does not exist, but it looks like the handling/logic for it is broken anyways. This debug change is now removed .

**Pull request merged**